pkg/controller/service/handlers.go

```diff
@@ -2168,6 +2168,16 @@ func (s *Service) getDevice(w http.ResponseWriter, r *http.Request) {
 								return
 							}
 
+							deviceServiceStates, err := s.deviceServiceStates.GetDeviceServiceStates(
+								r.Context(), project.ID, device.ID, application.ID)
+							if err == nil {
+								applicationStatusInfo.ServiceStates = deviceServiceStates
```

Comment:
Are you sure this won't run be nil and cause an error on the frontend?

Reply:
It actually, will, but the frontend handles this properly at the moment. I'll fix in a follow up since it's not super relevant to this.